cloudtools/scripts/aws_watch_pending.py

```diff
@@ -306,7 +306,15 @@ def do_request_instance(region, moz_instance_type, price, ami, instance_config,
         groups=instance_config[region].get("security_group_ids"))
     nc = NetworkInterfaceCollection(spec)
 
-    user_data = user_data_from_template(moz_instance_type, fqdn, region)
+    if 'user_data_file' in instance_config[region]:
```

Comment:
Instead of having conditions here, can you refactor get_user_data_tmpl() https://github.com/mozilla/build-cloud-tools/blob/60fc80ac75726b9dcefedc86a48f90f0f2aeed1c/cloudtools/aws/__init__.py#L145 so it's more generic? I'd rename *.cloud-init files to *.user-data and used it for windows too. No conditions, no "user_data_file", less headache with debugging.

Another option would be explicitly set and pass user-data template file names to that function.